lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -19,120 +20,194 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+
+Function CheckModified-File($url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy) {
+
+    $fileLastMod = ([System.IO.FileInfo]$dest).LastWriteTimeUtc
+    $webLastMod = $null
+
+    $webRequest = [System.Net.HttpWebRequest]::Create($url)
+
+    foreach ($header in $headers.GetEnumerator()) {
+        $webRequest.Headers.Add($header.Name, $header.Value)
+    }
+
+    if ($timeout) {
+        $webRequest.Timeout = $timeout * 1000
+    }
+
+    if (-not $use_proxy) {
+        # Ignore the system proxy settings
+        $webRequest.Proxy = $null
+    } elseif ($proxy) {
+        $webRequest.Proxy = $proxy
+    }
+
+    if ($credentials) {
+        $webRequest.Credentials = $credentials
+    }
+
+    $webRequest.Method = "HEAD"
+    Try {
+        [System.Net.HttpWebResponse]$webResponse = $webRequest.GetResponse()
+
+        $webLastMod = $webResponse.GetResponseHeader("Last-Modified")
+    } Catch [System.Net.WebException] {
+        $result.status_code = $_.Exception.Response.StatusCode
+        Fail-Json -obj $result -message "Error requesting '$url'. $($_.Exception.Message)"
+    } Catch {
+        Fail-Json -obj $result -message "Error when requesting 'Last-Modified' date from '$url'. $($_.Exception.Message)"
+    }
+    $result.status_code = $webResponse.StatusCode
+    $result.msg = $webResponse.StatusDescription
+    $webResponse.Close()
+
+    if ($webLastMod -and ((Get-Date -Date $webLastMod) -lt $fileLastMod)) {
+        return $false
+    } else {
+        return $true
+    }
+}
+
+
+Function Download-File($result, $url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy, $whatif) {
+
+    # Check $dest parent folder exists before attempting download, which avoids unhelpful generic error message.
+    $dest_parent = Split-Path -LiteralPath $dest
+    if (-not (Test-Path -LiteralPath $dest_parent -PathType Container)) {
+        Fail-Json -obj $result -message "The path '$dest_parent' does not exist for destination '$dest', or is not visible to the current user.  Ensure download destination folder exists (perhaps using win_file state=directory) before win_get_url runs."
+    }
+
+    $webClient = New-Object System.Net.WebClient
+
+    foreach ($header in $headers.GetEnumerator()) {
+        $webClient.Headers.Add($header.Name, $header.Value)
+    }
+
+    if ($timeout) {
+        $webClient.Timeout = $timeout * 1000
```

Comment:
Ditto for timeout

Reply:
So here we do have a WebClient object. It's strange that it worked fine for me, I'll remove the timeout here.